RetailCoder.VBE/UI/Refactorings/AssignedByValParameterQuickFixDialog.cs

```diff
@@ -1,22 +1,22 @@
 ﻿using System;
-using System.Linq;
 using System.Windows.Forms;
 using Rubberduck.Parsing.Symbols;
 using Rubberduck.Inspections;
+using Rubberduck.Inspections.QuickFixes;
+using Rubberduck.VBEditor;
 
 namespace Rubberduck.UI.Refactorings
 {
     public partial class AssignedByValParameterQuickFixDialog : Form, IDialogView
     {
-        private readonly string[] _procedureLines;
-        private bool _userInputIsValid;
+        private AssignedByValParameterQuickFixHelper _quickFixHlpr;
 
-        public AssignedByValParameterQuickFixDialog(string[] procedureLines)
+        public AssignedByValParameterQuickFixDialog(Declaration target, QualifiedSelection selection)
         {
-            _procedureLines = procedureLines;
-            _userInputIsValid = false;
             InitializeComponent();
             InitializeCaptions();
+            Target = target;
+            _quickFixHlpr = new AssignedByValParameterQuickFixHelper(target, selection);
```

Comment:
why not reuse the helper instance? Why does the dialog need the helper instance in the first place? the decoupling of presentation and business logic seems to lack a bit here :)